framework/contracts/native/ibc-host/src/account_commands.rs

```diff
@@ -80,18 +80,22 @@ pub fn receive_register(
 pub fn receive_dispatch(
     _deps: DepsMut,
     account: AccountBase,
-    manager_msg: manager::ExecuteMsg,
+    manager_msgs: Vec<manager::ExecuteMsg>,
 ) -> HostResult {
     // execute the message on the manager
-    let manager_call_msg = wasm_execute(account.manager, &manager_msg, vec![])?;
+    let msgs = manager_msgs
+        .into_iter()
+        .map(|msg| wasm_execute(&account.manager, &msg, vec![]))
+        .collect::<Result<Vec<_>, _>>()?;
 
-    // We want to forward the data that this execution gets
-    let submsg = SubMsg::reply_on_success(manager_call_msg, RESPONSE_REPLY_ID);
+    let response = Response::new().add_attribute("action", "receive_dispatch");
 
-    // Polytone handles all the necessary
-    Ok(Response::new()
-        .add_submessage(submsg)
-        .add_attribute("action", "receive_dispatch"))
+    // If there is only one executed message, we want to forward the data that this execution gets
+    Ok(if msgs.len() == 1 {
+        response.add_submessage(SubMsg::reply_on_success(msgs[0].clone(), RESPONSE_REPLY_ID))
+    } else {
+        response.add_messages(msgs)
+    })
```

Comment:
I believe it can be useful on multi-action as well, shouldn't it just forward data of the last message?

Reply:
Agreed that it can be useful. Do you think it's something the user expects ?

Reply:
Should we do just like Polytone does, collecting all collected data ? 
https://github.com/DA0-DA0/polytone/blob/main/contracts/main/proxy/src/contract.rs#L73
We could do the same on account action !

Reply:
> Should we do just like Polytone does, collecting all collected data ? https://github.com/DA0-DA0/polytone/blob/main/contracts/main/proxy/src/contract.rs#L73 We could do the same on account action !

Looks good

Reply:
Ok, forwarding if there are multiple messages.
Forwarding the whole data will be in another release, creating tasks for it

Reply:
Created [ABS-400](https://linear.app/abstract-sdk/issue/ABS-400/allow-forwarding-all-the-data-from-executed-messages) to address the multiple response data issue.